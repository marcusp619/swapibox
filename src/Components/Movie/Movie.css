.star-wars {
  display: flex;
  justify-content: center;
  height: 800px;
  /*This sets allows us to transform the text on a 3D plane, and is somewhat a magic number*/
  perspective: 400px;
  color: #111111;
  font-family: 'Poller One', cursive;
  font-size: 100%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.crawl {
  position: relative;
  /*top: -100px;*/
  transform-origin: 50% 100%;
  /* Adds the crawl animation, which plays for one minute */
  animation: crawl 60s linear;
}

.Movie {
  height: 400px;
  padding: 0 10em;
}

@keyframes crawl {
  0% {
    /* The element starts below the screen */
    top: 0;
    font-size: 300%;
    line-height: 100%;
    /* Rotate the text 20 degrees but keep it close to the viewer */
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    /* This is a magic number, but using a big one to make sure the text is fully off the screen at the end */
    top: -6000px;
    font-size: 300%;
    line-height: 100%;
    /* Slightly increasing the rotation at the end and moving the text far away from the viewer */
    transform: rotateX(25deg) translateZ(-2500px);
  }
}