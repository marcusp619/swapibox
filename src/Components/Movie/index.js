import React from 'react';
import './Movie.css';
import './StarWarsIntro.css'

const Movie = ({ film }) => {
  return (
    <section className="star-wars Movie">
      <div className="crawl">
        <p>{ film }</p>
      </div>
    </section>
  );
};

export default Movie;
