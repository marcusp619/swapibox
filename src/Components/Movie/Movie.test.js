import React from 'react';
import ReactDOM from 'react-dom';
import Movie from './';
import { shallow } from 'enzyme';

let wrapper

describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Movie />);
  });
})
  
beforeEach(() => {
  wrapper = shallow(<Movie />);
});

it('should match snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});

