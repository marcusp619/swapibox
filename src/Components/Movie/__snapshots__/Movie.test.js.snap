// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<section
  className="star-wars Movie"
>
  <div
    className="crawl"
  >
    <p />
  </div>
</section>
`;
